subsys/bluetooth/host/id.c

```diff
@@ -1282,15 +1282,26 @@ int bt_id_create(bt_addr_le_t *addr, uint8_t *irk)
 	int new_id, err;
 
 	if (addr && !bt_addr_le_eq(addr, BT_ADDR_LE_ANY)) {
-		if (addr->type != BT_ADDR_LE_RANDOM ||
-		    !BT_ADDR_IS_STATIC(&addr->a)) {
-			LOG_ERR("Only static random identity address supported");
-			return -EINVAL;
-		}
-
 		if (id_find(addr) >= 0) {
 			return -EALREADY;
 		}
+
+		if (addr->type == BT_ADDR_LE_PUBLIC && !IS_ENABLED(CONFIG_BT_PRIVACY)) {
```

Comment:
Privacy doesn't really have anything to do with this. Even if privacy is enabled, the *identity address* can still be a public address. So that condition should be removed, I think.

That said, would be good to double-check that we don't have any code which assumes the identity address is always a static random one.

Reply:
I'm afraid I don't really understand how bluetooth privacy works then, I've removed the check though.

Reply:
> I'm afraid I don't really understand how bluetooth privacy works then, I've removed the check though.

Thanks. Devices always have an identity address, which can be either public or static random. Even with privacy enabled, the identity address gets shared with peers during pairing, together with the Identity Resolving Key.